sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidFieldAccessCheck.java

```diff
@@ -0,0 +1,198 @@
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import com.github.sevntu.checkstyle.SevntuUtil;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>
+ * Checks that certain fields are not used. This can be used to enforce that fields like e.g.
+ * {@link java.util.Locale#ROOT}.
+ * </p>
+ *
+ * <p>
+ * Parameters are:
+ * </p>
+ *
+ * <ul>
+ * <li><b>packageName</b> - The field package name to be forbidden.</li>
+ * <li><b>className</b> - The field class name to be forbidden.</li>
+ * <li><b>fieldName</b> - The field name to be forbidden.</li>
+ * </ul>
+ *
+ * <p>
+ * Together, these three parameters forms the field fully qualified name.
+ * </p>
+ *
+ * <p>
+ * Default parameters are:
+ * </p>
+ *
+ * <ul>
+ * <li><b>packageName</b>java.util</li>
+ * <li><b>className</b>Locale</li>
+ * <li><b>fieldName</b>ROOT</li>
+ * </ul>
+ *
+ * <p>
+ * which forbids the usage of {@link java.util.Locale#ROOT}.
+ * </p>
+ *
+ * @author <a href="mailto:yasser.aziza@gmail.com">Yasser Aziza</a>
+ * @since 1.38.0
+ */
+public class ForbidFieldAccessCheck extends AbstractCheck {
+
+    /**
+     * Warning message key.
+     */
+    public static final String MSG_KEY = "forbid.field.access";
+
+    /**
+     * '.' character used as separator between FQN elements.
+     */
+    private static final char DOT = '.';
+
+    /**
+     * Package name.
+     */
+    private String packageName = "java.util";
+
+    /**
+     * Class name.
+     */
+    private String className = "Locale";
+
+    /**
+     * Field name.
+     */
+    private String fieldName = "ROOT";
+
+    /**
+     * Whether the field class was imported.
+     */
+    private boolean wasPackageImported;
+
+    /**
+     * Sets the package name, in which the field is declared.
+     * @param packageName the field name
+     */
+    public void setPackageName(String packageName) {
+        this.packageName = packageName;
+    }
+
+    /**
+     * Sets the class name, which declares the field.
+     * @param className the class name
+     */
+    public void setClassName(String className) {
+        this.className = className;
+    }
+
+    /**
+     * Sets the Field name, which should be forbidden to use.
+     * @param fieldName the field name
+     */
+    public void setFieldName(String fieldName) {
+        this.fieldName = fieldName;
+    }
+
+    /**
+     * Gets the field fully qualified name.
+     * @return {@link String} containing the field FQN
+     */
+    private String getFieldFullyQualifiedName() {
+        return packageName + DOT + className + DOT + fieldName;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.STATIC_IMPORT,
+            TokenTypes.IMPORT,
+            TokenTypes.IDENT,
```

Comment:
You need to view PACKAGE_DEF as the class being looked for can be in the same package.